torchgen/utils.py

```diff
@@ -97,6 +97,15 @@ def context(msg_fn: Callable[[], str]) -> Iterator[None]:
         raise
 
 
+if TYPE_CHECKING:
```

Comment:
I'm not sure if the BC checker is "TYPE_CHECKING" or not... but if this passes the BC checker then I'm okay with it.

Reply:
`typing_extensions.assert_never` provides the exact same function.